core/arch/arm/mm/core_mmu_v7.c

```diff
@@ -671,8 +671,10 @@ static void map_memarea(struct tee_mmap_region *mm, uint32_t *ttb)
 	 * TODO: support mapping devices at a virtual address which isn't
 	 * the same as the physical address.
 	 */
-	if (mm->va < (NUM_UL1_ENTRIES * SECTION_SIZE))
-		panic("va conflicts with user ta address");
+	if (ttb == (void *)core_mmu_get_main_ttb_va()) {
```

Comment:
> IRAM is small and page mapping is not allowed.

Indeed 16kByte of level1 MMU table is IRAM consuming. Note however that a single applying a small page mapping over & single virtual address range (1MByte wide) range mapping through 4kB only consumes one extra kByte of MMU table. 
Note that if you want to split in ROM/RAM/XROM, you will definitely need small page mapping.

In case of LPAE support, MMU table have other sizes. Consider a 4kB table for the level1 MMU table, and an eventual extra 4kB table for the level2 MMU table (if small page mapping is required).

You expect this alternate mapping to be loaded at core initialization, or at runtime, when suspend/resume functions are called ?